app.ts

```diff
@@ -0,0 +1,406 @@
+								</div>`);
+						}
+					} else {
+						$('.results-box').remove();
+					}
+				})
+				.catch(error => {
+					throw new Error(`Failed when attempting to search: ${error}`);
+				});
+		});
+
+		// Will take the range on the DOM and return records based on that range.
+		$('.heading').on('click', '.results-select', (event: any) => {
+			$('.results-select').prop('disabled', true);
+			let startID: number;
+			let endID: number;
+			let pageEnd: number;
+			let pageStart: number;
+			return this.getRecordCount()
+				.then(count => {
+					let idRange = $('.results-select').text();
+					let rangeArray = null;
+					rangeArray = idRange.split('-');
+					$('.results-box').remove();
+					startID = parseInt(rangeArray[0]);
+					endID = parseInt(rangeArray[1]);
+					if (!isNaN(endID) && Number.isInteger(endID)) {
+						let maxRecords = this.recordsPerPage();
+						this.currentPage = Math.floor((endID + 1) / (maxRecords + 1));
+						pageEnd = Math.ceil(this.currentPage / 10) * 10;
+						pageStart = pageEnd - 9;
+						if (endID >= count) {
+							startID = ((this.currentPage - 1) * maxRecords) + 1;
+							endID = (count - 1);
+							this.paginationStart = pageStart;
+							this.paginationEnd = pageEnd;
+						}
+					} else {
+						throw new Error("Please provide a valid integer.");
+					}
+				})
+				.then(() => {
+					return this.pageNumbers(pageStart, pageEnd);
+				})
+				.then(() => {
+					return this.showRecords(startID, endID);
+				})
+				.then(() => {
+					$('.results-select').prop('disabled', false);
+				})
+				.catch(error => {
+					throw new Error(`Failed when clicking on the search range: ${error}`);
+				});
+		});
+	}
+
+	/** Will calculate the amount records to be shown according to the screen height. */
+	adjustDisplayedRecords(): Promise<void> {
+		let screenHeight = <number>($('#records-table-body').height());
+		let pageStart: number;
+		let pageEnd: number;
+		return this.getRecordCount()
+			.then(count => {
+				let maxRecords = this.recordsPerPage();
+				let inputNumber = this.input();
+				if (inputNumber === -1) {
+					let newCurrentPage = Math.ceil(this.currentFromID / maxRecords);
+					if (newCurrentPage === 0) {
+						this.currentFromID = 0;
+						newCurrentPage = 1;
+					}
+					this.currentToID = this.currentFromID + maxRecords;
+					this.currentPage = newCurrentPage;
+					let originalID = ((this.currentPage * maxRecords) + (this.currentPage - 1)) - maxRecords;
+					this.difference = originalID - this.currentFromID;
+				} else {
+					if (this.currentToID >= count) {
+						this.currentToID = (count - 1);
+					}
+					let newCurrentPage = Math.ceil(inputNumber / maxRecords);
+					this.currentToID = newCurrentPage * maxRecords;
+					this.currentPage = newCurrentPage;
+					this.currentFromID = (this.currentPage - 1) * maxRecords + 1;
+				}
+				if (this.currentFromID === 0) {
+					let divisor = screenHeight < 136 ? maxRecords : maxRecords - 1;
```

Comment:
I don't like this hard coded value... 136. You definitely should not need it. Something must be wrong with the way you calculate `maxRecords` if this is necessary. Please remove it.